dfir_lang/src/graph/ops/fold_keyed.rs

```diff
@@ -152,41 +152,42 @@ pub const FOLD_KEYED: OperatorConstraints = OperatorConstraints {
             quote_spanned! {op_span=>
                 #assign_hashtable_ident
 
-                #work_fn(|| {
+                {
                     #[inline(always)]
-                    fn check_input<Iter, K, V>(iter: Iter) -> impl ::std::iter::Iterator<Item = #root::util::PersistenceKeyed::<K, V>>
+                    fn check_input<St, K, V>(st: St) -> impl #root::futures::stream::Stream<Item = #root::util::PersistenceKeyed::<K, V>>
                     where
-                        Iter: ::std::iter::Iterator<Item = #root::util::PersistenceKeyed::<K, V>>,
+                        St: #root::futures::stream::Stream<Item = #root::util::PersistenceKeyed::<K, V>>,
                         K: ::std::clone::Clone,
                         V: ::std::clone::Clone,
                     {
-                        iter
+                        st
                     }
 
                     /// A: accumulator type
                     /// T: iterator item type
-                    /// O: output type
                     #[inline(always)]
-                    fn call_comb_type<A, T, O>(a: &mut A, t: T, f: impl Fn(&mut A, T) -> O) -> O {
-                        (f)(a, t)
+                    fn call_comb_type<A, T>(a: &mut A, t: T, f: impl Fn(&mut A, T)) {
+                        let () = (f)(a, t);
```

Comment:
Nice catch

Reply:
It was actually intentional IIRC, to allow `Lattice::merge` which returns `bool` to be used directly, but I think we should move away from that fuzzy typing